src/waveform/renderers/waveformrendererpreroll.cpp

```diff
@@ -76,8 +76,14 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
         if (preRollVisible) {
             // VSample position of the right-most triangle's tip
             double triangleTipVSamplePosition =
-                    numberOfVSamples * playMarkerPositionFrac -
+                    numberOfVisibleVSamples * playMarkerPositionFrac -
                     currentVSamplePosition;
+            double invisibleVSamples = triangleTipVSamplePosition - numberOfVisibleVSamples;
+            if (invisibleVSamples > 0) {
+                triangleTipVSamplePosition -=
+                        floor(invisibleVSamples / polyVSampleOffset) *
+                        polyVSampleOffset;
+            }
```

Comment:
Could you please add/refine some comments, for example
`// VSample position of the right-most triangle's tip just before the track start`
`// samples on the right of the viewport boundary`
`// get tip position of the last visible triangle to draw only visible triangles`
(pretty much the same, only adjusted wording, for the postroll area)